### PR TITLE
build: update gateway-api

### PR DESCRIPTION
**Pull request by michaelbeaumont**
Rebased our fork on a newer version that includes https://github.com/kubernetes-sigs/gateway-api/pull/1354

### Checklist prior to review

- [x] Link to docs PR or issue --
- [x] Link to UI issue or PR --
- [x] Is the [issue worked on linked][1]? --
- [x] The PR does not hardcode values that might break projects that depend on kuma (e.g. "kumahq" as a image registry) --
- [x] The PR will work for both Linux and Windows, system specific functions like `syscall.Mkfifo` have equivalent implementation on the other OS --
- [x] Unit Tests --
- [x] E2E Tests --
- [x] Manual Universal Tests --
- [x] Manual Kubernetes Tests --
- [x] Do you need to update [`UPGRADE.md`](/UPGRADE.md)? --
- [x] Does it need to be backported according to the [backporting policy](/CONTRIBUTING.md#backporting)? --

[1]: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

### CODE REVIEW

File: pkg/plugins/runtime/k8s/controllers/gatewayapi/gateway_conversion.go [hunk 1]
```diff
@@ -210,42 +211,57 @@ func (r *GatewayReconciler) gapiToKumaGateway(
...
+					if kube_apierrs.IsNotFound(err) {
+						name := fmt.Sprintf("%q %q", policyRef.GroupKindReferredTo().String(), policyRef.NamespacedNameReferredTo().String())
+						unresolvableCertRef = &badCertRef{
+							reason:  string(gatewayapi.ListenerReasonInvalidCertificateRef),
+							message: fmt.Sprintf("invalid reference to %s", name),
+						}
+						break
+					} else {
+						return nil, nil, err
+					}
```
Comment (bartsmykla) [inline pkg/plugins/runtime/k8s/controllers/gatewayapi/gateway_conversion.go:245]: I don't think there is need to have an else branch here
```suggestion
					}
					
					return nil, nil, err
```

**Review by bartsmykla — APPROVED** [review]
one nit, but overall lgtm

File: pkg/plugins/runtime/k8s/controllers/gatewayapi/gateway_conversion.go [hunk 2]
```diff
@@ -210,42 +211,57 @@ func (r *GatewayReconciler) gapiToKumaGateway(
...
 				}
+
+				referencedSecrets = append(referencedSecrets, policyRef.NamespacedNameReferredTo())
 			}
 
-			if len(unresolvableCertRefs) == 0 {
+			if unresolvableCertRef == nil {
```
Comment (jakubdyszkiewicz) [inline pkg/plugins/runtime/k8s/controllers/gatewayapi/gateway_conversion.go:251]: this method has 182 lines and growing. Could we maybe extract cert handling to a separate method?

Comment (jakubdyszkiewicz) [reply pkg/plugins/runtime/k8s/controllers/gatewayapi/gateway_conversion.go]: nice, thank you

**Review by jakubdyszkiewicz — APPROVED** [review]

**Pull request merged**